WordPress/Classes/ViewRelated/Gutenberg/EditorMediaUtility.swift

```diff
@@ -5,12 +5,14 @@ import Gridicons
 final class AuthenticatedImageDownload: AsyncOperation {
     let url: URL
     let blog: Blog
+    private let callbackQueue: DispatchQueue
     private let onSuccess: (UIImage) -> ()
     private let onFailure: (Error) -> ()
 
-    init(url: URL, blog: Blog, onSuccess: @escaping (UIImage) -> (), onFailure: @escaping (Error) -> ()) {
+    init(url: URL, blog: Blog, callbackQueue: DispatchQueue, onSuccess: @escaping (UIImage) -> (), onFailure: @escaping (Error) -> ()) {
```

Comment:
Originally the `onSuccess` closure is called on the main queue, and the `onFailure` closure is called on a background thread that runs this operation. Adding this `calbackQueue` to unify their calling queue. Also, it makes the enforcing callback queue code in `MediaThumbnailService` nicer, without requiring additional `async` calls there.